lib/season_statistics.rb

```diff
@@ -0,0 +1,52 @@
+class SeasonStatistics
+    attr_reader :teams, :game_teams, :games, :game_statistics
+  
+    def initialize(teams, game_teams, games)
+      @teams = teams
+      @game_teams = game_teams
+      @games = games
+      @game_statistics = GameStatistics.new(@games)
```

Comment:
So this was the culprit huh? Wish I could have had the foresight to help you figure this out. Well done sticking with it @JonoSommers .